src/sentry/static/sentry/app/components/charts/transparentLoadingMask.tsx

```diff
@@ -1,12 +1,24 @@
+import React from 'react';
 import styled from '@emotion/styled';
 
 import LoadingMask from 'app/components/loadingMask';
 
 type Props = {
   visible: boolean;
-};
+  className?: string;
+  children?: React.ReactNode;
+} & React.HTMLProps<HTMLDivElement>;
 
-const TransparentLoadingMask = styled(LoadingMask)<Props>`
+const TransparentLoadingMask = styled(
+  ({className, visible, children, ...props}: Props) => {
+    const other = visible ? {...props, 'data-test-id': 'loading-placeholder'} : props;
+    return (
+      <LoadingMask className={className} {...other}>
+        {children}
+      </LoadingMask>
+    );
+  }
+)<Props>`
```

Comment:
Ugh, I hate how noisy this can get when we're mostly just trying to add a prop. Wonder if we should add something like this https://github.com/emotion-js/emotion/issues/821#issuecomment-546204409

Reply:
Yes it is a lot of ceremony for conditionally adding one property.